### PR TITLE
feat(pack.migration): add contact phone on resume view for migration

### PR DESCRIPTION
**Pull request by Steffy29**
| Question         | Answer
| ---------------- | ---
| Branch?          | `develop`
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | Fix #UXCT-299
| License          | BSD 3-Clause

## Description

Add contact on resume view for migration offer

### CODE REVIEW

File: packages/manager/apps/telecom/src/app/telecom/pack/migration/confirm/pack-migration-confirm.html [hunk 1]
```diff
@@ -156,6 +156,51 @@ <h3 data-translate="telecom_pack_migration_doing"></h3>
...
+                    data-translate="telecom_pack_migration_confirm_resume_contact_phone"
+                >
+                </label>
+                <input
+                    type="text"
+                    class="form-control"
+                    id="contactPhone"
+                    name="contactPhone"
+                    data-ui-validate="{ pattern: '$ctrl.checkContactPhone($value)' }"
+                    data-ng-model="$ctrl.contactPhone"
```
Comment (marie-j) [inline packages/manager/apps/telecom/src/app/telecom/pack/migration/confirm/pack-migration-confirm.html:186]: Why not directly use `$ctrl.process.contactPhone` as a model ?

File: packages/manager/apps/telecom/src/app/telecom/pack/migration/confirm/pack-migration-confirm.html [hunk 2]
```diff
@@ -156,6 +156,51 @@ <h3 data-translate="telecom_pack_migration_doing"></h3>
...
+                    class="control-label"
+                    data-translate="telecom_pack_migration_confirm_resume_contact_phone"
+                >
+                </label>
+                <input
+                    type="text"
+                    class="form-control"
+                    id="contactPhone"
+                    name="contactPhone"
+                    data-ui-validate="{ pattern: '$ctrl.checkContactPhone($value)' }"
```
Comment (marie-j) [inline packages/manager/apps/telecom/src/app/telecom/pack/migration/confirm/pack-migration-confirm.html:185]: I don't see any `checkContactPhone` method in the controller am I missing something ?

File: packages/manager/apps/telecom/src/app/telecom/pack/migration/confirm/pack-migration-confirm.html (hunk 1 above)
Comment (marie-j) [inline packages/manager/apps/telecom/src/app/telecom/pack/migration/confirm/pack-migration-confirm.html:186]: Why not directly
```suggestion
                    data-ng-model="$ctrl.process.contactPhone"
```
?

File: packages/manager/apps/telecom/src/app/telecom/pack/migration/confirm/pack-migration-confirm.html [hunk 3]
```diff
@@ -156,6 +156,51 @@ <h3 data-translate="telecom_pack_migration_doing"></h3>
...
+            class="widget-presentation-title"
+            data-translate="telecom_pack_migration_confirm_resume_contact_owner"
+        ></h2>
+        <div class="d-inline-flex p-2 bd-highlight">
+            <div
+                class="form-group"
+                data-ng-class="{
+                    'has-error': resumeForm.contactPhone.$touched && resumeForm.contactPhone.$invalid
+                }"
+            >
```
Comment (marie-j) [inline packages/manager/apps/telecom/src/app/telecom/pack/migration/confirm/pack-migration-confirm.html:173]: Can't all this be switched to a `oui-field` ?

File: packages/manager/modules/telecom-universe-components/src/telecom/pack/migration/pack-migration-process.service.js [hunk 4]
```diff
@@ -161,6 +161,13 @@ export default /* @ngInject */ function($q, OvhApiPackXdsl, Poller) {
       });
     }
 
+    // Set contact phone if is set
+    if (migrationProcess.contactPhone) {
+      assign(postParams, {
+        contactPhone: migrationProcess.contactPhone,
+      });
```
Comment (marie-j) [inline packages/manager/modules/telecom-universe-components/src/telecom/pack/migration/pack-migration-process.service.js:168]:
```suggestion
      Object.assign(postParams, {
        contactPhone: migrationProcess.contactPhone,
      });
```
lodash doesn't seem necessary here

**Review by marie-j — DISMISSED** [review]

**Comment by antleblanc** [comment]
@Steffy29 Could you please resolve existing conflicts? Thank you!

**Comment by Steffy29** [comment]
@antleblanc  conflicts are resolved

**Review by marie-j — APPROVED** [review]

**Review by antleblanc — DISMISSED** [review]

**Review by JeremyDec — DISMISSED** [review]

**Review by mohammed-zahaf — APPROVED** [review]

File: packages/manager/modules/telecom-universe-components/src/telecom/pack/migration/pack-migration-process.service.js [hunk 5]
```diff
@@ -216,8 +223,29 @@ export default /* @ngInject */ function($q, OvhApiPackXdsl, Poller) {
...
+    return OvhApiPackXdsl.v6()
+      .getContactOwner({
+        packName: migrationProcess.pack.packName,
+      })
+      .$promise.then((result) => {
+        migrationProcess.contactOwner = result.data;
+        if (migrationProcess.contactOwner) {
+          migrationProcess.contactPhone = migrationProcess.contactOwner.phone;
+        }
+      });
```
Comment (marie-j) [inline packages/manager/modules/telecom-universe-components/src/telecom/pack/migration/pack-migration-process.service.js:240]: I think it is now possible to use : 
```suggestion
          migrationProcess.contactPhone = migrationProcess.contactOwner?.phone;
```

**Review by marie-j — DISMISSED** [review]

**Review by JeremyDec — DISMISSED** [review]

**Review by mohammed-zahaf — DISMISSED** [review]

File: packages/manager/apps/telecom/src/app/telecom/pack/migration/confirm/pack-migration-confirm.html [hunk 6]
```diff
@@ -4,266 +4,288 @@
...
-            data-translate="telecom_pack_migration_confirm_migrate_to"
-            data-translate-values="{ fromOffer : $ctrl.process.pack.offerDescription, toOffer : $ctrl.process.selectedOffer.description }"
-        ></p>
+            <p
+                data-translate="telecom_pack_migration_confirm_migrate_to"
+                data-translate-values="{ fromOffer : $ctrl.process.pack.offerDescription, toOffer : $ctrl.process.selectedOffer.description }"
+            ></p>
+
+            <div>
+                <div class="responsive-table-line my-3">
```
Comment (marie-j) [inline packages/manager/apps/telecom/src/app/telecom/pack/migration/confirm/pack-migration-confirm.html:20]: Not using a oui-datagrid ?

Comment (Steffy29) [reply packages/manager/apps/telecom/src/app/telecom/pack/migration/confirm/pack-migration-confirm.html:20]: Nope. Like we already discussed on this subject a time ago, an oui-datagrid componant could not be applied here.

**Review by antleblanc — APPROVED** [review]

**Pull request merged**